### PR TITLE
Add lti_user_identites migration

### PR DESCRIPTION
**Pull request by nicklathe**
This PR is the first of two that adds a new LTI User Identity model. It contains only:

* schema.rb changes
* the migration file

This model creates two associations:

* `belongs_to :lti_integrations`
* `belongs_to :users`

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/db/migrate/20230522232402_create_lti_user_identities.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateLtiUserIdentities < ActiveRecord::Migration[6.1]
+  def change
+    create_table :lti_user_identities do |t|
+      t.string :subject, null: false
+      t.references :lti_integration, null: false, foreign_key: true, type: :bigint
+      t.references :user, null: false, foreign_key: true, type: :integer
```
Comment (elf-code) [inline dashboard/db/migrate/20230522232402_create_lti_user_identities.rb:6]: Any reason why you re-ordered the columns relative to the Tech Spec? https://docs.google.com/document/d/1tTwAF40cnn2oUvK2CmRO-a5qYXkCR5rR5gBJS2-zV_U/edit#heading=h.ulde8t7jj1tf

**Review by elf-code — APPROVED** [review]
Looks good, we might want to warn the oncall that there's more migrations coming down the pipeline

**Review by elf-code — APPROVED** [review]
Looks good, we might want to warn the oncall that there's more migrations coming down the pipeline

**Review by carl-codeorg — APPROVED** [review]

File: dashboard/db/migrate/20230522232402_create_lti_user_identities.rb (hunk 1 above)
Comment (nicklathe) [reply dashboard/db/migrate/20230522232402_create_lti_user_identities.rb:6]: Not particularly. I'm not sure they are ordered well in the tech spec though? We'll usually query these on `subject`, and it's the only non-foreign key column, so it felt like it should come first. The other two are 🤷‍♂️

Comment (carl-codeorg) [reply dashboard/db/migrate/20230522232402_create_lti_user_identities.rb:6]: Is subject going to be the same ID from the `sub` claim from the JWT?

Comment (nicklathe) [reply dashboard/db/migrate/20230522232402_create_lti_user_identities.rb:6]: Yep. Initially I had this as `lti_user_id`, but figured aligning with the `subject` from the JWT was more clear.

**Pull request merged**